distributed/dashboard/components/scheduler.py

```diff
@@ -1720,6 +1724,336 @@ def __del__(self):
+                sort_dependents = [ele for ele in stack_order if ele in dependents[tg]]
+
+                for dep in sort_dependents:
+                    if dep in y:
+                        continue
+                    else:
+                        y[dep] = (
+                            y[tg] + self.height_node * (len(dependents[tg]) - 1) / 2
+                        )
+
+                if (x[tg], y[tg]) in collision:
+
+                    old_x, old_y = x[tg], y[tg]
+                    x[tg], y[tg] = collision[(x[tg], y[tg])]
+
+                    y[tg] += self.height_node * scale
+                    collision[old_x, old_y] = (x[tg], y[tg])
+                else:
+                    collision[(x[tg], y[tg])] = (x[tg], y[tg])
+
+                # info neded for node layout to coulmn data source
+                nodes_layout[tg] = {}
+                nodes_layout[tg]["x"] = x[tg]
+                nodes_layout[tg]["y"] = y[tg]
+
+                # info needed for arrow layout
+                arrows_layout[tg] = {}
+                arrows_layout[tg]["nstart"] = dependencies[tg]
+                arrows_layout[tg]["nend"] = [tg] * len(dependencies[tg])
+
+            return nodes_layout, arrows_layout
+
+    @without_property_validation
+    def update(self):
+
+        if not self.scheduler.task_groups:
+            self.subtitle.text = "Scheduler is empty."
+        else:
+            self.subtitle.text = " "
+
+        if self.nodes_layout.keys() != self.scheduler.task_groups.keys():
+            self.nodes_layout, self.arrows_layout = self.update_layout()
+
+        nodes_data = {
+            "x": [],
+            "y": [],
+            "name": [],
+            "name_short": [],
+            "color": [],
+            "tot_tasks": [],
+            "x_start": [],
+            "x_end": [],
+            "y_start": [],
+            "y_end": [],
+            "x_end_progress": [],
+            "progress": [],
+            "x_label": [],
+            "y_label": [],
+            "mem_alpha": [],
+            "node_line_width": [],
+            "xc_pbar": [],
+            "yc_pbar": [],
+            "comp_tasks": [],
+        }
+
+        arrows_data = {
+            "xs": [],
+            "ys": [],
+            "xe": [],
+            "ye": [],
+        }
+
+        for key, tg in self.scheduler.task_groups.items():
+            x = self.nodes_layout[key]["x"]
+            y = self.nodes_layout[key]["y"]
+
+            comp_tasks = (
+                tg.states["waiting"] + tg.states["no-worker"] + tg.states["processing"]
+            )
+            tot_tasks = sum(tg.states.values())
+
+            # main boxes layout
+            nodes_data["x"].append(x)
+            nodes_data["y"].append(y)
+
+            nodes_data["name"].append(tg.prefix.name)
+            nodes_data["name_short"].append(
+                tg.prefix.name[:10]
+            )  # This needs to be different
```

Comment:
For longer task prefix names, we should truncate (as you've already done) and add `...` to indicate the name has been truncated. We use the following logic in the progress plot today:

https://github.com/dask/distributed/blob/6340b5b34a582ff1ffe3eaf9dd7f3aac33d0a983/distributed/diagnostics/progress_stream.py#L93

Using the same logic here seems sensible